pkg/steps/template.go

```diff
@@ -521,84 +518,65 @@ func waitForPodCompletionOrTimeout(ctx context.Context, podClient coreclientset.
 	//   the state of the pod is correctly reported, artifacts are gathered,
 	//   and termination happens deterministically for both success and failure
 	//   scenarios.
-	watcher, err := podClient.Watch(context.Background(), meta.ListOptions{
```

Comment:
Why did we lose this watch?

Reply:
Because its not supported by the client to open a raw watch. We could use a cache-backed client but I am not sure its worth the effort. This has the nice sideffect that writing a warpper is possible and writing tests is easier

Reply:
Wait so what is the replacement for the watch? I am confused

Reply:
There is no replacement for the watch on the client level because you are not supposed to hand-manage watches. You can make the client itself cache-backed in which case reading requests will get served from a cache that is in turn backed by a standard SharedIndexInformer